js/ad_manager_vast.js

```diff
@@ -64,6 +64,69 @@ OO.Ads.manager(function(_, $) {
+    var iframe                          = null;
+    var timeline                        = [];
+
+    // ad settings
+    var transitionFromNonLinearVideo    = false;
+    var adLoaded                        = false;
+    var adRequestType                   = 'adRequest';
+
+    // VPAID variables
+    this._slot                          = null;
+    this._videoSlot                     = null;
+    this._videoSlotCanAutoPlay          = true;
+    this.environmentVariables           = {};
+    this._eventsCallbacks               = {};
+    this._parameters                    = {};
+
+    var VPAID_EVENTS                    = {
+                                            AD_LOADED                 : 'AdLoaded',
+                                            AD_STARTED                : 'AdStarted',
+                                            AD_STOPPED                : 'AdStopped',
+                                            AD_SKIPPED                : 'AdSkipped',
+                                            AD_SKIPPABLE_STATE_CHANGE : 'AdSkippableStateChange',
+                                            AD_DURATION_CHANGE        : 'AdDurationChange',
+                                            AD_SIZE_CHANGE            : 'AdSizeChange',
+                                            AD_LINEAR_CHANGE          : 'AdLinearChange',
+                                            AD_INTERACTION            : 'AdInteraction',
+                                            AD_IMPRESSION             : 'AdImpression',
+                                            AD_CLICK_THRU             : 'AdClickThru',
+                                            AD_PAUSED                 : 'AdPaused',
+                                            AD_PLAYING                : 'AdPlaying',
+                                            AD_VIDEO_START            : 'AdVideoStart',
+                                            AD_VIDEO_COMPLETE         : 'AdVideoComplete',
+                                            AD_ERROR                  : 'AdError',
+                                            AD_LOG                    : 'AdLog',
+                                            AD_REMAINING_TIME_CHANGE  : 'AdRemainingTimeChange',
+                                            AD_VIDEO_FIRST_QUARTILE   : 'AdVideoFirstQuartile',
+                                            AD_VIDEO_MIDPOINT         : 'AdVideoMidpoint',
+                                            AD_VIDEO_THIRD_QUARTILE   : 'AdVideoThirdQuartile',
+                                            AD_USER_ACCEPT_INVITATION : 'AdUserAcceptInvitation',
+                                            AD_VOLUME_CHANGE          : 'AdVolumeChange',
+                                            AD_USER_MINIMIZE          : 'AdUserMinimize'
+                                          };
+
+    // If `false` prerolls won't load until intialPlay
+    this.preload                          = true;
```

Comment:
In general we want to not preload ads. This also adds another hook into VAST to get ads to preload. So ideally we want to consolidate this into 1.